src/core/Akka/Actor/ActorCell.Children.cs

```diff
@@ -372,7 +378,7 @@ private IInternalActorRef MakeChild(Props props, string name, bool async, bool s
 
             // In case we are currently terminating, fail external attachChild requests
             // (internal calls cannot happen anyway because we are suspended)
-            if (ChildrenContainer.IsTerminating)
+            if (IsFastTerminating() || ChildrenContainer.IsTerminating)
```

Comment:
`IsFastTerminating()` checks a state flag I added to indicate that this `ActorCell` is terminating. Added it to solve #2024, where long story short: the immutable collection operations happening inside the constructor of the `ChildContainer` take too long and result in the `ChildContainer.IsTerminated` check on `ActorCell.MakeChild` always winning the race.

Adding this flag turned that check into a constant time operation and eliminated speediness of the `IChildContainer` as a potential race condition.